inquire/src/ansi.rs

```diff
@@ -0,0 +1,92 @@
+use std::str::Chars;
+
+#[must_use]
+enum MatchResult<'a> {
+    Matched { remaining_chars: Chars<'a> },
+    NotMatched,
+}
+
+/// Match/consume the next ANSI escape code if it exists.
+/// The format is based on [this description](https://handwiki.org/wiki/ANSI_escape_code#CSI_(Control_Sequence_Introducer)_sequences)
+fn match_escape_code(chars: Chars<'_>) -> MatchResult<'_> {
```

Comment:
Is this cross-platform? Should we use different rules depending on the platform? From a quick search looks like Windows ANSI and Standard ANSI differ

Reply:
I think it is, but I'll need to research more before being sure.